docs/stories/components/Checkbox.stories.js

```diff
@@ -25,21 +51,112 @@ function CheckboxStory() {
           value={val}
         />
       </div>
+      <div style={{ marginTop: '-5px', marginBottom: 20 }}>
+        <ul>
+          <Checkbox
+            name="selectAll"
+            onChange={handleChange}
+            message={message}
+            someChecked={hasSomeCheckboxSelected && !areAllCheckboxesSelected}
+            value={areAllCheckboxesSelected}
+          />
+          {Object.keys(checkboxes).map(current => {
+            return (
+              <li
+                key={current}
+                style={{ marginBottom: 1, paddingTop: 18, paddingLeft: 15 }}
+              >
+                <Checkbox
+                  name={current}
+                  message={current}
+                  value={checkboxes[current]}
+                  onChange={({ target: { name, value } }) => {
+                    setCheckboxValue(prevState => ({
+                      ...prevState,
+                      [name]: value,
+                    }));
+                  }}
+                />
+              </li>
+            );
+          })}
+        </ul>
+      </div>
       <Pre>
         {`
 import React, { useState } from 'react';
 import { Checkbox } from '@buffetjs/core';
 
 function Example() {
-  const [value, setValue] = useState(false);
+  const [val, setValue] = useState(false);
+  const [checkboxes, setCheckboxValue] = useState({
+    checkbox1: false,
+    checkbox2: true,
+  });
+  const defaultProps = {
+    message: 'Check me',
+    name: 'checkbox',
+  };
+
+  const areAllCheckboxesSelected = Object.keys(checkboxes).every(
+    key => checkboxes[key] === true
+  );
+
+  const hasSomeCheckboxSelected = Object.keys(checkboxes).some(
+    key => checkboxes[key] === true
+  );
+
+  const handleChange = () => {
+    const valueToSet = !areAllCheckboxesSelected;
+
+    setCheckboxValue(prevState => {
+      return Object.keys(prevState).reduce((acc, current) => {
+        acc[current] = valueToSet;
+
+        return acc;
+      }, {});
+    });
+  };
+
+  const message = areAllCheckboxesSelected ? 'Unselect all' : 'Select all';
 
   return (
-    <Checkbox
-      message="Check me"
-      name="checkbox"
-      onChange={({ target }) => setValue(target.value)}
-      value={value}
-    />
+    <>
+      <Checkbox
+        message="Check me"
+        name="checkbox"
+        onChange={({ target }) => setValue(target.value)}
+        value={value}
```

Comment:
value doesn't exist

`value={val}`

Reply:
Updated the name of the variable instead